CONTRIBUTING.md

```diff
@@ -1,192 +1,180 @@
+### Testing changes
 
-    ./gradlew wrapper --gradle-distribution-url=file:/path/to/gradle-<version>-all.zip
+After making changes, you can test your code in 2 ways:
 
-### Development Setup
+1. Run tests.
+- Run `./gradlew :<subproject>:quickTest` where `<subproject>` is the name of the subproject you've changed. 
+- For example: `./gradlew :launcher:quickTest`.
+2. Install Gradle locally and try out a change in behavior manually. 
+- Install: `./gradlew install -Pgradle_installPath=/any/path`
+- Use: `/any/path/bin/gradle taskName`.
 
-In order to make changes to Gradle, you'll need:
+It's also a good idea to run `./gradlew sanityCheck` before submitting your change because this will help catch code style issues.
 
-* A text editor or IDE. We use and recommend [IntelliJ IDEA CE](http://www.jetbrains.com/idea/).  IntelliJ Ultimate will also work.
-* A [Java Development Kit](http://jdk.java.net/) (JDK) version 11.
-* [git](https://git-scm.com/) and a [GitHub account](https://github.com/join).
+### Submitting Your Change
 
-Gradle uses pull requests for contributions. Fork [gradle/gradle](https://github.com/gradle/gradle) and clone your fork. Configure your Git username and email with:
+After you submit your pull request, a Gradle developer will review it. It is normal for this to take several iterations, so don't get discouraged by change requests. They ensure the high quality that we all enjoy.
 
-    git config user.name 'First Last'
-    git config user.email user@example.com
+If you need to check on [CI](http://builds.gradle.org/) status as an external contributor, you can login as "guest".
 
-### IntelliJ
+## Useful tips
 
-You require IntelliJ 2018.3.1 or newer.
-- Open the `build.gradle.kts` file with IntelliJ and choose "Open as Project"
-- Make sure "Create separate module per source set" is selected
-- Make sure  "Use default gradle wrapper" is selected
-- Select a Java 11 VM as "Gradle JVM"
-- In the "File already exists" dialogue, choose "Yes" to overwrite
-- In the "Open Project" dialogue, choose "Delete Existing Project and Import"
-- Revert the Git changes to files in the `.idea` folder
+### Debugging Gradle
 
-NOTE: Due to the project size, the initial import can take a while and IntelliJ might become unresponsive for several seconds during this period.
+You can debug Gradle by adding `-Dorg.gradle.debug=true` to the command-line. Gradle will wait for you to attach a debugger at `localhost:5005` by default.
 
-IntelliJ automatically hides stacktrace elements from the `org.gradle` package, which makes running/debugging tests more difficult.  You can disable this behavior by changing IntelliJ Preferences under Editor -> General -> Console. In the "Fold lines that contain" section, remove the `org.gradle` entry.
+If you made changes to build logic in the `build-logic` included build, you can run its tests by executing `./gradlew :build-logic:check`.
```

Comment:
This is actually broken locally.... we need to fix this. https://gradle.slack.com/archives/CBSJ2GUTV/p1649677454431979